src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -258,7 +609,7 @@ internal TensorFlowTransform(IHostEnvironment env, TFSession session, string[] i
             {
                 var tfOutput = new TFOutput(Graph[Outputs[i]]);
                 var shape = Graph.GetTensorShape(tfOutput);
-                int[] dims = shape.NumDimensions > 0 ? shape.ToIntArray().Skip(shape[0] == -1 ? BatchSize : 0).ToArray() : new[] { 0 };
+                int[] dims = shape.NumDimensions > 0 ? shape.ToIntArray().Skip(shape[0] == -1 ? 1 : 0).ToArray() : new[] { 0 };
```

Comment:
>1 [](start = 96, length = 1)

Wonder why? if we feed, let's say 4 rows to input, we would get only 1 output? or actually 4?

Reply:
Here we want to skip the first dimension (which is the batch dimension) and `1` means the first one. If we set it to `BatchSize` and `BatchSize=10` then it will skip 10 elements.

---
In reply to: [221771362](https://github.com/dotnet/machinelearning/pull/1063#discussion_r221771362) [](ancestors = 221771362)